lib/utils.js

```diff
@@ -15,13 +15,42 @@ module.exports.extractKVBindings = (setFlagValues) => {
     const key = kvComponents[0]
     const value = kvComponents.slice(1).join('=')
 
+    let filepath
+
+    if (setFlagPaths) {
+      // Find a variable path from the paths which matches the variable
+      const variablePath = setFlagPaths.find(value => {
+        if (value.split('=').length < 2) {
+          throw new Error('Invalid kv-file flag format. Expected format of [variable]=[value]')
+        }
+        return (value.indexOf(variable + '=') > -1)
+      })
+
+      // If there is is a variable matching, then get the path name from it
+      if (variablePath) filepath = variablePath.split('=')[1]
+    }
+
     if (!bindings[variable]) {
-      bindings[variable] = new KeyValueStore()
+      bindings[variable] = new KeyValueStore(filepath)
     }
 
     bindings[variable].put(key, value)
   }
 
+  if (setFlagPaths.length > 0 && setFlagValues.length == 0) {
+    for (const flag of setFlagPaths) {
```

Comment:
Use the object based on my comment above here instead of parsing the flag paths again.

Reply:
Good idea - resolved now